src/span.js

```diff
@@ -137,24 +137,51 @@ export default class Span {
     }
 
     /**
+     * Checks whether or not a span can be written to.
+     *
+     * @return {boolean} - The decision about whether this span can be written to.
+     **/
+    _isWriteable(): boolean {
+        // A span can NOT be written to if it:
+        // 1.) is finished.
+        // 2.) has been finalized, and is not sampled
+        return  this._duration === undefined &&
+                (!this._spanContext.samplingFinalized ||
+                this._spanContext.isSampled());
+    }
+
+    /**
      * Sets the operation name on this given span.
      *
      * @param {string} name - The name to use for setting a span's operation name.
      * @return {Span} - returns this span.
      **/
     setOperationName(operationName: string): Span {
+        if (!this._isWriteable()) {
+            return this;
+        }
+
         this._operationName = operationName;
         // If the span has debug set, then the question of resampling
         // after setting the operation name is irrelevant.  The span is always
         // sampled.
-        if (!this._spanContext.isDebug()) {
+
+        // Do not re-sample the span if:
+        // 1.)  Debug has been set on the span.  Sampling is irrelevant in this case
+        //      the span will be reported no matter what.
+        // 2.)  The span is the first in process.  If the span is the first in process
+        //      then the sampling decision has already been made.
+        // 3.)  The span is not writable.  This can happen if the span's context was used
+        //      to create another span, or if the span is finished.
+        if (!this._spanContext.isDebug() || !this._firstInProcess) {
```

Comment:
I m not convinced about this condition. First let's rewrite the comment as suggested above. Then, why does "first in process" need to matter here? IIRC "first in process" was used in other clients to avoid sending process-level tags in the span. The new data model sends process-level tags at the batch level, so I am not sure fi we even need "first in process". Either way though, it is very different from the sampling - what we want for sampling is an indicator that the sampling decision has been inherited from the parent span. The easiest way to do that is probably to set `finalized` to true when creating such span.